src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -206,7 +208,7 @@ private void AllocateWriteHeadSynchronized(int sizeHint)
             }
         }
 
-        private BufferSegment AllocateSegment(int sizeHint)
+        private BufferSegment AllocateSegment(int minimumSegmentSize, int sizeHint)
```

Comment:
Nit: We should rename the `sizeHint` to `minSize` or something in internal/private APIs because it now violates the contract to provide spans/memory smaller than that so it's not really a "hint". With the new variables, this becoming somewhat confusing.

Reply:
Yes we should but I dislike unrelated changes. I'll make that change as well.